src/docs/components/properties.md

```diff
@@ -692,6 +692,105 @@ export class ToDoListItem {
 }
 ```
 
+#### Mutable Arrays and Objects
+
+Stencil compares Props by reference in order to efficiently rerender components.
+Setting `mutable: true` on a Prop that is an object or array allows the _reference_ to the Prop to change inside the component and trigger a render.
+It does not allow a mutable change to an existing object or array to trigger a render. 
```

Comment:
Possible wording suggestion that may flow a bit more smoothly?

```suggestion
Setting `mutable: true` on a Prop that is an object or array will cause Stencil to trigger a re-render whenever the _reference_ to the Prop changes inside the component.
However, mutating an existing object in-place will not trigger a re-render, since the reference to that object will not change. A common
pattern for forcing a re-render in this situation is to create a new array or object instead of mutating the existing instance in place.
```

Reply:
Also a general question here, I think using a compare-by-reference check w/ `==` or `===` and rendering only when there's a new object (thereby avoiding any deep equality checks or anything silly like that) is a pretty common pattern in JS, are there any good explanatory articles we could link out to that break this down well? I checked MDN and didn't find a great explanation of compare-by-reference vs compare-by-value. Maybe we can assume a Stencil reader already knows this?

Reply:
cool, that all sounds fine to me, especially since I didn't find something that did a great job breaking it down with a little googling

Reply:
👍 anything else you want to see here in that case? (PR is still marked as 'Changes Requested')